File: example/bench/fs_read.py
import os
import time


def main():
    path = os.environ.get("BENCH_PATH", "/tmp/bench.txt")
    iterations = int(os.environ.get("BENCH_ITER", "500"))
    payload = "x" * 1024
    errors = 0

    try:
        with open(path, "w", encoding="utf-8") as handle:
            handle.write(payload)
    except Exception as exc:
        print(f"[Bench] write failed: {exc}")

    start = time.perf_counter()
    for _ in range(iterations):
        try:
            with open(path, "r", encoding="utf-8") as handle:
                handle.read()
        except Exception:
            errors += 1
    elapsed_ms = (time.perf_counter() - start) * 1000

    print(f"[Bench] fs_read elapsed_ms={elapsed_ms:.2f} errors={errors}")


if __name__ == "__main__":
    main()

File: example/bench/fs_read.ts
const path = process.env.BENCH_PATH ?? "/tmp/bench.txt";
const iterations = Number(process.env.BENCH_ITER ?? "500");
const payload = "x".repeat(1024);
let errors = 0;

try {
  await Bun.write(path, payload);
} catch (error) {
  console.warn(`[Bench] write failed: ${error}`);
}

const start = performance.now();
for (let i = 0; i < iterations; i += 1) {
  try {
    await Bun.file(path).text();
  } catch {
    errors += 1;
  }
}
const elapsedMs = performance.now() - start;

console.log(`[Bench] fs_read elapsed_ms=${elapsedMs.toFixed(2)} errors=${errors}`);
